src/mindlessgen/qm/xtb.py

```diff
@@ -71,13 +79,22 @@ def optimize(
             # read the optimized molecule
             optimized_molecule = molecule.copy()
             optimized_molecule.read_xyz_from_file(temp_path / "xtbopt.xyz")
-
+            if np.any(np.isin(molecule.ati, get_lanthanides())):
+                # Reset the UHF value to the original value before returning the optimized molecule.
+                optimized_molecule.uhf = uhf_original
             return optimized_molecule
 
     def singlepoint(self, molecule: Molecule, verbosity: int = 1) -> str:
         """
         Perform a single-point calculation using xtb.
         """
+        if np.any(np.isin(molecule.ati, get_lanthanides())):
+            if check_ligand_uhf(molecule.ati, molecule.charge):
```

Comment:
Also this `if` statement is quite obsolete then. Just call the `check_...` function and continue with the rest of the code.